rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -245,16 +247,19 @@ func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endB
 					return fmt.Errorf("fatal: validateBatch failed: finalize event: %v, err: %w", event, err)
 				}
 
-				if index == batchIndex {
-					rawdb.WriteFinalizedL2BlockNumber(s.db, endBlock)
-					rawdb.WriteLastFinalizedBatchIndex(s.db, batchIndex)
-					log.Info("write finalized l2 block number", "batch index", index, "finalized l2 block height", endBlock)
-				}
 				rawdb.WriteFinalizedBatchMeta(s.db, index, finalizedBatchMeta)
 
 				if index%100 == 0 {
 					log.Info("finalized batch progress", "batch index", index, "finalized l2 block height", endBlock)
 				}
+
+				highestFinalizedBlockNumber = endBlock
+			}
+
+			if index == batchIndex {
```

Comment:
is it possible to break this condition?

Reply:
if this condition is broken (seems impossible but it can be considered one kind of inconsistency), we can either `panic`, or let it run into the next loop and crash due to batch hash mismatch?

Reply:
you are right, I will remove this check